app/lib/attachment_batch.rb

```diff
@@ -78,9 +78,17 @@ def remove_files
             logger.debug { "Deleting #{attachment.path(style)}" }
 
             begin
+              retries ||= 0
               attachment.send(:directory).files.new(key: attachment.path(style)).destroy
-            rescue Fog::Storage::OpenStack::NotFound
-              # Ignore failure to delete a file that has already been deleted
+            rescue => e
+              if e.is_a?(Fog::OpenStack::Storage::NotFound)
+                logger.debug { "Will ignore because file is not found #{attachment.path(style)}" }
+              else
+                sleep(5)
+                retry if (retries += 1) < 3
```

Comment:
How about re-using the `MAX_RETRY` constant here? The default value is also `3` but can be overriden via an ENV variable (which is prefixed with `S3_` and this is not S3, I know, still might be worthwile?).

Reply:
You are correct, better than hardcoding the number of retries.